packages/strapi-admin/admin/src/containers/Webhooks/EditView/utils/formatData.js

```diff
@@ -0,0 +1,40 @@
+import { set } from 'lodash';
+
+const cleanData = data => {
+  const webhooks = data;
+  const headers = cleanHeaders(data.headers);
+
+  set(webhooks, 'headers', unformatHeaders(headers));
+  return webhooks;
+};
+
+const unformatHeaders = headers => {
+  return headers.reduce((obj, item) => {
+    const { key, value } = item;
+    return {
+      ...obj,
+      [key]: value,
+    };
+  }, {});
+};
+
+const cleanHeaders = headers => {
+  if (Object.keys(headers).length === 1) {
+    const { key, value } = headers[0];
+    if (key.length === 0 && value.length === 0) {
+      return [];
+    }
+  }
+  return headers;
+};
+
+const cleanErrors = errors => {
```

Comment:
I'm not sure you need this logic you can simply use your `errorMessage` (which will be `getErrorMessage` directly.